core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/transports/util/XsdUtil.java

```diff
@@ -190,14 +191,15 @@ public static void printXsd(CarbonHttpRequest request,
                     outputStream.flush();
                 }
             } else if  (xsds.endsWith(".xsd") && xsds.indexOf("..") == -1){
-                InputStream in = axisService.getClassLoader()
-                        .getResourceAsStream(DeploymentConstants.META_INF + "/" + xsds);
-                if (in != null) {
-                    outputStream.write(IOUtils.getStreamAsByteArray(in));
-                    outputStream.flush();
-                    outputStream.close();
-                } else {
-                    response.setError(HttpServletResponse.SC_NOT_FOUND);
+                try (InputStream in = axisService.getClassLoader()
```

Comment:
What will happens to the catch?

Reply:
The try was introduced in order to automatically close the input stream after the code block is executed by using the try-with-resources[1] statement and not for an error handling purpose.

The possible exception that can be thrown by the code segment inside the try block here is an IO exception. This exception is already present in the method signature[2], therefore if the exception occurs, it will be thrown to the object which is invoking the method.

Therefore, a catch is not needed in this situation.

[1] https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
[2] https://github.com/wso2/carbon-kernel/blob/4.9.x/core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/transports/util/XsdUtil.java#L49